pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -106,56 +130,91 @@ def check_checksum(path: Path, pkg: Dict[str, Any]):
         raise ValueError("Invalid {} checksum".format(checksum_algorithm))
 
 
+def trim_archive_extension(tarballname):
+    for extension in [
+        ".tar.gz",
+        ".tgz",
+        ".tar",
+        ".tar.bz2",
+        ".tbz2",
+        ".tar.xz",
+        ".txz",
+        ".zip",
+    ]:
+        if tarballname.endswith(extension):
+            return tarballname[: -len(extension)]
+    return tarballname
+
+
 def download_and_extract(
-    buildpath: Path, packagedir: Path, pkg: Dict[str, Any], args
+    buildpath: Path, srcpath: Path, src_metadata: Dict[str, Any], args
 ) -> Path:
-    srcpath = buildpath / packagedir
+    """
+    Download the source from specified in the meta data, then checksum it, then
+    extract the archive into srcpath.
 
-    if "source" not in pkg:
-        return srcpath
+    buildpath -- The path to the build directory. Generally will be
```

Comment:
Same, please use numpydoc formatting for docstrings https://numpydoc.readthedocs.io/en/latest/format.html#parameters (without specifying types since it's already in the type annotations)

Here and in all functions where a docstring was changed.